coprocessor/fhevm-engine/fhevm-listener/src/cmd/mod.rs

```diff
@@ -375,10 +375,9 @@ impl InfiniteLogIter {
                 }
             }
         }
-        let Some(current_event) = &self.current_event else {
-            return None;
+        if self.current_event.is_some() else {
+            self.last_block_event_count += 1;
         };
-        self.last_block_event_count += 1;
         return self.current_event.clone();
```

Comment:
If self.current_event is none, existing code returns None without incrementing last_block_event_count and new code does. Does that have any side effects?